src/compiler/types.ts

```diff
@@ -5100,14 +5100,16 @@ export interface TypeChecker {
     getDefaultFromTypeParameter(type: Type): Type | undefined;
 
     /** @internal */ getAnyType(): Type;
-    /** @internal */ getStringType(): Type;
-    /** @internal */ getNumberType(): Type;
-    /** @internal */ getBooleanType(): Type;
-    /** @internal */ getFalseType(fresh?: boolean): Type;
-    /** @internal */ getTrueType(fresh?: boolean): Type;
+    getStringType(): Type;
+    getStringLiteralType(value: string): StringLiteralType;
+    getNumberType(): Type;
+    getNumberLiteralType(value: number): NumberLiteralType;
+    getBooleanType(): Type;
+    getFalseType(fresh?: boolean): Type;
+    getTrueType(fresh?: boolean): Type;
     /** @internal */ getVoidType(): Type;

```

Comment:
Is there any particular reason to not export the other ones in this block, like any, void, never, etc?

Reply:
`any`, `void`, `never`, and `symbol` seem fine. The others are either a bit implementation-detaily or could be affected by library types.

Reply:
Yep, those were the 4 I meant (I should have listed them out). The others ("optional type"), definitely not,